applications/services/gui/modules/number_input.c

```diff
@@ -0,0 +1,493 @@
+#include "number_input.h"
+
+#include <gui/elements.h>
+#include <furi.h>
+#include <assets_icons.h>
+
+/** NumberInput type */
+struct NumberInput {
+    View* view;
+};
+
+typedef struct {
+    const char text;
+    const uint8_t x;
+    const uint8_t y;
+} NumberInputKey;
+
+typedef struct {
+    const char* header;
+    FuriString* text_buffer;
+    bool clear_default_text;
+    int32_t max_value;
+    int32_t min_value;
+    bool useSign;
+    char sign;
```

Comment:
Too many redundant things do you really need them?

Reply:
Please explain, for me these are not redundant, they are all needed. 
max_value => highest number allowed
min_value => lowest number allowed
sign => + or - depending if the number is positive or negative
useSign => if true, will show the button that allows changing from positive to negative numbers 
(In my use cases so far I did not want negative numbers and therefore don't want that button, this was a change request from your side)